### PR TITLE
add generic linux pin class using libgpiod python bindings

### PR DESCRIPTION
**Pull request by brennen**
**WORK IN PROGRESS, DO NOT MERGE**

### CODE REVIEW

File: src/adafruit_blinka/microcontroller/generic_linux/pin.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        return str(self.id)
+
+    def __eq__(self, other):
+        return self.id == other
+
+    def init(self, mode=IN, pull=None):
+        if mode != None:
+            if mode == self.IN:
+                flags = ()
+                if pull != None:
```
Comment (brennen) [inline src/adafruit_blinka/microcontroller/generic_linux/pin.py]: Is this section at all on the right track?

**Comment by ladyada** [comment]
i think this looks good - just need to make sure it works :)

**Comment by brennen** [comment]
Yeah, I've tested a little bit with just a button, but I was planning to plug it into the Pi class and check against some real code.

**Comment by ladyada** [comment]
try gutting out the RPi.GPIO code we have now, see if it works

**Comment by ladyada** [comment]
@brennen is this ready for testing/mergin?

**Comment by ladyada** [comment]
i cherrypicked this for my orangepi port - turns out this isnt a good choice for raspi because it doesnt have pullup support. so we'll stick with rpi.gpio for now. closing but your code lives on https://github.com/adafruit/Adafruit_Blinka/pull/59 !!!

**Pull request closed without merging**